docs/reference/faq/index.md

```diff
@@ -131,4 +211,651 @@ for more details.
+PoolingHttpClientConnectionManager expires.For more details,
+see [here](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html)
+
+#### What is logRequest config?
+
+If the response received from the http sink endpoint is null or if the status code of the response is within the range
+of **SINK_HTTP_REQUEST_LOG_STATUS_CODE_RANGES** config, then the request payload is logged. e;g, `200-500`
+
+#### What is shouldRetry config?
+
+If the response received from the http sink endpoint is null or if the status code of the response is within the range
+of **SINK_HTTP_RETRY_STATUS_CODE_RANGES** config, then the request is again retried for a configured number of attempts
+with backoff. e;g, `400-500`
+
+#### What happens in case of null response?
+
+If the request to the sink fails with a null status code, Firehose will attempt to retry calling the endpoint for a
+configured number of attempts with backoff. After that, if DLQ is enabled, the messages are pushed to DLQ queue with
+backoff. If not enabled, no more attempts are made.
+
+#### What happens in case of null status code in a non-null response?
+
+If the request to the sink fails with a null status code, Firehose will drop the messages and not retry sending them
+again.
+
+#### When is a message dropped?
+
+If the request to the sink fails with a retryable status code as specified via **SINK_HTTP_RETRY_STATUS_CODE_RANGES**
+config, Firehose will attempt to retry calling the endpoint for a configured number of attempts with backoff. After
+that, if DLQ is enabled, the messages are pushed to DLQ queue with backoff. If DLQ is disabled, messages are dropped.
+
+#### What is the difference between Parameterised vs Dynamic Url ?
+
+In parameterised query and parameterised header HTTP sink mode, the incoming Kafka message is converted to query
+parameters and header parameters in the HTTP request respectively. In case of dynamic url HTTP sink mode, the input
+Kafka message is parsed into the request body.
+
+#### For parameterised header, how is the data added?
+
+1. set **SINK_HTTP_PARAMETER_SOURCE** to either `key` or `message`
+   (based on whether the field one requires in the http request is part of the key or the message in the Kafka record),
+2. set **SINK_HTTP_PARAMETER_PLACEMENT** to `header`, and
+3. set **SINK_HTTP_PARAMETER_SCHEMA_PROTO_CLASS** to the Protobuf class for the input Kafka message and **
+   INPUT_SCHEMA_PROTO_TO_COLUMN_MAPPING** to a json value indicating the proto field number from the input message to be
+   mapped to the header name.
+
+### General Queries
+
+#### Which Java versions firehose work with?
+
+Firehose has been developed and tested to work with Java 8. Compatibility with lower or higher versions is not known and
+hence, needs to be tested prior to use.
+
+#### How does the execution work? Is the entire process sync or async ?
```

Comment:
@Meghajit I think this is outdated.. do yo wanna mention async consumer too?

Reply:
@lavkesh Added a reference to the AsyncConsumer documentation.